var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -156,6 +156,20 @@ def configure_args(self):
 
         return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
+    def configure(self, spec, prefix):
+        # Run the default autotools package configure
+        super(Hdf5, self).configure(spec, prefix)
+
+        if '@:1.8.14' in spec:
```

Comment:
We should probably continue this in another thread but I pretty much agree with @adamjstewart on all of this.

@adamjstewart: if you want tot to the arg injection thing, look in bin/spack... I added a really basic version of it to allow commands to take an optional argument  for any args unknown to the parser.  It's used for `spack test`.

This stuff makes a lot of sense when you're writing hooks, which is essentially what the package file is doing.